### PR TITLE
Rename signature algorithm ECDSA_Secp256k1 to ECDSA_secp256k1

### LINKED ISSUE #760: Inconsistent SigningAlgorithm identifiers between flow-go and cadence repositories.
### Instructions

Please fill out the template below to the best of your ability and include a label indicating which tool/service you were working with when you encountered the problem.

### Problem

On the currently running testnet, when creating a PublicKey that uses `secp256k1` I get an `unkown error` with a pointer to the algorithm identifier `SignatureAlgorithm.ECDSA_secp256k1`.

```
let publicKey = PublicKey(
    publicKey: "1234".decodeHex(),
    signatureAlgorithm: SignatureAlgorithm.ECDSA_secp256k1
)
```

### Steps to Reproduce

Send this transaction with a valid secp256k1 public key to testnet.

```
transaction(publicKey: String) {
  prepare(signer: AuthAccount) {
    let account = AuthAccount(payer: signer);
    let key = PublicKey(
      publicKey: publicKey.decodeHex(),
      signatureAlgorithm: SignatureAlgorithm.ECDSA_secp256k1
    );
    account.keys.add(
      publicKey: key,
      hashAlgorithm: HashAlgorithm.SHA3_256,
      weight: 1.0
    );
  }
}
```

### Acceptance Criteria

Not sure.

### Context

In the `flow-go` repository (flow-go/crypto/types.go) you have the following string identifiers.

```go
func (f SigningAlgorithm) String() string {
	return [...]string{"UNKNOWN", "BLS_BLS12381", "ECDSA_P256", "ECDSA_secp256k1"}[f]
}
```

My assumption, when building the transaction initially, was that these identifiers will be understood by Cadence. But for some reason Cadence only accepts `SignatureAlgorithm.ECDSA_Secp256k1`, emphasis on the uppercase **S**.

### PR DESCRIPTION
**Pull request by SupunS**
Closes #760

## Description

Work towards #582

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/types.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ const (
 	// Supported signing algorithms
 	SignatureAlgorithmUnknown         = sema.SignatureAlgorithmUnknown
 	SignatureAlgorithmECDSA_P256      = sema.SignatureAlgorithmECDSA_P256
-	SignatureAlgorithmECDSA_Secp256k1 = sema.SignatureAlgorithmECDSA_Secp256k1
+	SignatureAlgorithmECDSA_Secp256k1 = sema.SignatureAlgorithmECDSA_secp256k1
```
Comment (SupunS) [inline runtime/types.go:51]: I didn't change the interface type (this constant) so that this change won't affect FVM.

**Review by tarakby — APPROVED** [review]
👏

File: runtime/types.go (hunk 1 above)
Comment (turbolent) [reply runtime/types.go:51]: We should change it everywhere to be consistent here, it's also upper-case on the FVM side

**Review by turbolent — COMMENTED** [review]
Nice! 

This is a breaking change, we need to make sure that contracts on Testnet and Mainnet will be upgrable / upgraded

**Comment by SupunS** [comment]
That's a good point on updatability! 
Technically it should be OK since `SignatureAlgorithm.ECDSA_secp256k1` can never be used as a type (can only use as a value), so no risk of anyone using it as a field of a contract. 

At the storage level, only the raw value along with the type name `SignatureAlgorithm` is stored, and the case-name `ECDSA_secp256k1` is never stored. So that should be fine as well.

But yeah, we need to make sure they are properly updated.

**Pull request merged**